task_planning.py

```diff
@@ -0,0 +1,180 @@
+# Proof of Concept for a task planning and execution system using
+# OpenAIs Functions and topological sort, based on the idea in 
+# question_answer_subquery.py. 
+# This version is simplified and centralizes the execution logic
+# in contrast tu the recursive approach in question_answer_subquery.py
+
+import openai
+import asyncio
+from pydantic import Field
+from typing import List, Generator
+from openai_function_call import OpenAISchema
+
+
+class QueryAnswer(OpenAISchema):
+    query_id: int
+    answer: str
+
+
+class Query(OpenAISchema):
```

Comment:
would be cool to rewrite this as a Task and SubTask planner

Reply:
I changed the naming throughout to Tasks instead of queries